front/app/modules/commercial/analytics/admin/components/EmailDeliveriesCard/useEmailDeliveries/parse.ts

```diff
@@ -53,7 +59,12 @@ const parseRow = (
 };
 
 const getDate = (row: PreparedTimeSeriesResponseRow) => {
-  return moment(get(row, 'first_dimension_date_sent_date'));
+  const dateAttribute =
+    row.first_dimension_date_sent_date ||
+    row.first_dimension_date_sent_week ||
+    row.first_dimension_date_sent_month;
+
+  return moment(dateAttribute);
```

Comment:
@jamesspeake I looked at the typing here as well as you suggested. Could we just use this instead? Functionally it does the same thing you had, but it uses the attribute names directly :) There wouldn't be a case where a row has all 3 values would there?

```
  const dateAttribute =
    row.first_dimension_date_sent_date ||
    row.first_dimension_date_sent_week ||
    row.first_dimension_date_sent_month;
```